netbox/dcim/tests/test_models.py

```diff
@@ -590,6 +590,34 @@ def test_device_duplicate_names(self):
         device2.full_clean()
         device2.save()
 
+    def test_device_label(self):
+        device1 = Device(
+            site=Site.objects.first(),
+            device_type=DeviceType.objects.first(),
+            role=DeviceRole.objects.first(),
+            name=None,
+        )
+        self.assertEqual(device1.label, None)
+
+        device2 = Device(
+            site=Site.objects.first(),
+            device_type=DeviceType.objects.first(),
+            role=DeviceRole.objects.first(),
+            name='Test Device 2',
+        )
+        self.assertEqual(device2.label, 'Test Device 2')
+
+        virtual_chassis = VirtualChassis.objects.create(name='VC 1')
+        device3 = Device(
+            site=Site.objects.first(),
+            device_type=DeviceType.objects.first(),
+            role=DeviceRole.objects.first(),
+            name=None,
+            virtual_chassis=virtual_chassis,
+            vc_position=2,
+        )
+        self.assertEqual(device3.label, 'VC 1:2')
```

Comment:
Should have a test if set name and virtual_chassis that it returns the name, not the generated one.